src/lua/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+MESSAGE(STATUS "Searching for Lua library --\n")
+FIND_PACKAGE(Lua)
+IF(NOT LUA_FOUND)
+	MESSAGE(FATAL_ERROR
+		"You need the Lua development library to be able to compile with Lua support.\nSee INSTALL file for details and more info\n"
+	)
+ENDIF(NOT LUA_FOUND)
+
+ADD_DEFINITIONS(-DLUA)
+
+ADD_CUSTOM_TARGET(
+	lua_bindings
+	COMMAND ${LUA_BINARY} generate_bindings.lua
+	BYPRODUCTS ${CMAKE_SOURCE_DIR}/src/lua/catabindings.cpp
```

Comment:
Quoting the documentation:
> The target has no output file and is ALWAYS CONSIDERED OUT OF DATE even if the commands try to create a file with the name of the target. Use ADD_CUSTOM_COMMAND to generate a file with dependencies.

And indeed, every time I run `make`, it recreates catabindings.cpp and recompiles and relinks it, which is at least a waste of time.

The standard Makefile has "$(LUA_DIR)/class_definitions.lua" and "$(LUASRC_DIR)/generate_bindings.lua" as explicit dependencies for regenerating the bindings.